src/librustc_codegen_llvm/debuginfo/metadata.rs

```diff
@@ -1794,13 +1801,15 @@ fn prepare_enum_metadata(
                 .as_generator()
                 .variant_range(enum_def_id, cx.tcx)
                 .map(|variant_index| {
-                    let name = SmallCStr::new(&substs.as_generator().variant_name(variant_index));
+                    let name = substs.as_generator().variant_name(variant_index);
                     unsafe {
                         Some(llvm::LLVMRustDIBuilderCreateEnumerator(
                             DIB(cx),
-                            name.as_ptr(),
-                            // FIXME: what if enumeration has i128 discriminant?
-                            variant_index.as_usize() as u64,
+                            name.as_ptr().cast(),
+                            name.len(),
+                            // Generators use u32 as discriminant type.
+                            variant_index.as_usize() as i64,
```

Comment:
```suggestion
                            variant_index.index().try_into().expect("Generators must use `u32` as the discriminant type"),
```

If we're removing the FIXME, let's use `expect`.

Reply:
Replaced with `as_u32().into()`